File: 2017/mixit-wasm/fibo.js
var seen = new Set();

var wasmImports = {
  env: {
    print_dom: function print_dom (key, value) {
        if (!seen.has(key)) {
            log(`fibo(${key}) = ${value}`);
            seen.add(key);
        }
    }
  }
};

WebAssembly.compile(wasmCode)
.then(module => WebAssembly.instantiate(module, wasmImports))
.then(instance => {
    let final = instance.exports.fibo(30);
    log(`final result = ${final}`);
})

File: 2017/mixit-wasm/fibo.cpp
extern void print_dom(int key, int value);

int fibo(int n) {
  if (n <= 1)
    return n;

  int prevprev = fibo(n - 2);
  print_dom(n - 2, prevprev);

  int prev = fibo(n - 1);
  print_dom(n - 1, prev);

  return prevprev + prev;
}
